### PR TITLE
plan: let union plan not change the original type.

### PR DESCRIPTION
**Pull request by hanfei1991**

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -520,6 +520,14 @@ func (b *planBuilder) buildDistinct(child LogicalPlan, length int) LogicalPlan {
 	return agg
 }
 
+func joinFieldType(a, b *types.FieldType) *types.FieldType {
```
Comment (XuHuaiyu) [inline plan/logical_plan_builder.go]: add a comment for this.
getUnionFieldType be more explicit?

**Comment by zimulala** [comment]
LGTM

File: plan/logical_plan_builder.go (hunk 1 above)
Comment (hanfei1991) [inline plan/logical_plan_builder.go]: In MySQL, this is not only for union.

**Comment by hanfei1991** [comment]
@XuHuaiyu PTAL

**Pull request merged**